lib/torrent.js

```diff
@@ -849,7 +850,29 @@ class Torrent extends EventEmitter {
 
     this._debug('add incoming peer %s', peer.id)
 
-    this._peers[peer.id] = peer
+    this._registerPeer(peer)
+  }
+
+  _registerPeer (newPeer) {
+    newPeer.on('download', downloaded => {
+      if (this.destroyed) return
+      this.received += downloaded
+      this._downloadSpeed(downloaded)
+      this.client._downloadSpeed(downloaded)
+      this.emit('download', downloaded)
```

Comment:
Would not this `destroyed` check be the same one that is in line `858` ?

Reply:
I don't think so, because every time you call `emit` it will call the registered handlers. So in every `emit` there is a chance that the user might have destroyed the object.